flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/async/queue/WatermarkQueueEntry.java

```diff
@@ -19,31 +19,44 @@
 package org.apache.flink.streaming.api.operators.async.queue;
 
 import org.apache.flink.annotation.Internal;
+import org.apache.flink.streaming.api.operators.TimestampedCollector;
 import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.util.Preconditions;
 
-import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+
+import java.util.Collection;
 
 /**
  * {@link StreamElementQueueEntry} implementation for the {@link Watermark}.
  */
 @Internal
-public class WatermarkQueueEntry extends StreamElementQueueEntry<Watermark> implements AsyncWatermarkResult {
+class WatermarkQueueEntry<OUT> implements StreamElementQueueEntry<OUT> {
+	@Nonnull
+	private final Watermark watermark;
 
-	private final CompletableFuture<Watermark> future;
+	WatermarkQueueEntry(@Nonnull Watermark watermark) {
```

Comment:
ditto: `@Nonnull` issue.
I guess it is the common sense for `Nonnull`. Then it is only necessary to tag it if `Nullable`?